internal/gateway/receiver/pool.go

```diff
@@ -143,13 +148,15 @@ M:
 
 			var taskType string
 			var result error
+			var finalRev uint64
 			switch t := task.(type) {
 			case payloadTask:
 				result = receiver.SubmitPayload(t.leasePath, t.payload, t.digest, t.headerSize)
 				taskType = "payload"
 			case commitTask:
-				result = receiver.Commit(t.leasePath, t.oldRootHash, t.newRootHash, t.tag)
+				finalRev, result = receiver.Commit(t.leasePath, t.oldRootHash, t.newRootHash, t.tag)
 				taskType = "commit"
+				t.finalRevChan <- finalRev
```

Comment:
From now on we don't expect anymore to read from `finalRevChan`, hence we should close it.

close(t.finalRevChan)

Reply:
And I meant to **write** not to read.